src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -172,77 +212,158 @@ private BoundPattern BindPropertyPattern(PropertyPatternSyntax node, BoundExpres
+                invoked: false,
+                diagnostics: diagnostics);
+
+            LookupResultKind resultKind = boundMember.ResultKind;
+            bool hasErrors = boundMember.HasAnyErrors || implicitReceiver.HasAnyErrors;
+
+            if (boundMember.Kind == BoundKind.PropertyGroup)
+            {
+                boundMember = BindIndexedPropertyAccess(
+                    (BoundPropertyGroup)boundMember, mustHaveAllOptionalParameters: true, diagnostics: diagnostics);
+                if (boundMember.HasAnyErrors)
                 {
-                    Error(diagnostics, ErrorCode.ERR_NoSuchMember, propName, type, propName.ValueText);
-                    pattern = new BoundWildcardPattern(node, hasErrors: true);
-                    result.Add(new BoundSubPropertyPattern(syntax, CreateErrorType(), null, resultKind, pattern, true));
+                    hasErrors = true;
                 }
             }
 
-            return result.ToImmutableAndFree();
+            ImmutableArray<BoundExpression> arguments = ImmutableArray<BoundExpression>.Empty;
+            ImmutableArray<string> argumentNamesOpt = default(ImmutableArray<string>);
+            ImmutableArray<int> argsToParamsOpt = default(ImmutableArray<int>);
+            ImmutableArray<RefKind> argumentRefKindsOpt = default(ImmutableArray<RefKind>);
+            bool expanded = false;
+
+            switch (boundMember.Kind)
+            {
+                case BoundKind.FieldAccess:
+                case BoundKind.PropertyAccess:
+                    break;
+
+                case BoundKind.IndexerAccess:
+                // TODO: Should a property pattern be capable of referencing an indexed property?
+                // https://github.com/dotnet/roslyn/issues/9375
+                //{
+                //    var indexer = (BoundIndexerAccess)boundMember;
+                //    arguments = indexer.Arguments;
+                //    argumentNamesOpt = indexer.ArgumentNamesOpt;
+                //    argsToParamsOpt = indexer.ArgsToParamsOpt;
+                //    argumentRefKindsOpt = indexer.ArgumentRefKindsOpt;
+                //    expanded = indexer.Expanded;
+                //    break;
+                //}
+
+                case BoundKind.DynamicIndexerAccess:
+                // TODO: Should a property pattern be capable of referencing a dynamic indexer?
+                // https://github.com/dotnet/roslyn/issues/9375
+                //{
+                //    var indexer = (BoundDynamicIndexerAccess)boundMember;
+                //    arguments = indexer.Arguments;
+                //    argumentNamesOpt = indexer.ArgumentNamesOpt;
+                //    argumentRefKindsOpt = indexer.ArgumentRefKindsOpt;
+                //    break;
+                //}
+
+                case BoundKind.EventAccess:
+                // TODO: Should a property pattern be capable of referencing an event?
+                // https://github.com/dotnet/roslyn/issues/9515
```

Comment:
It might make sense to allow referencing a field-like event from within a type that declares it (this is not a suggestion to handle this case in this change).